### PR TITLE
SCB-1374 support fsm redis channel

### PR DESCRIPTION
**Pull request by cmonkey**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean install` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

File: alpha/alpha-fsm/src/main/java/org/apache/servicecomb/pack/alpha/fsm/channel/AbstractActorEventChannel.java [hunk 1]
```diff
@@ -42,6 +45,7 @@ public void send(BaseEvent event) {
       this.sendTo(event);
       metricsService.metrics().doEventAccepted();
     } catch (Exception ex) {
+      logger.error("send Exception = [{}]", ex);
```
Comment (WillemJiang) [inline alpha/alpha-fsm/src/main/java/org/apache/servicecomb/pack/alpha/fsm/channel/AbstractActorEventChannel.java]: log error can show the stack trace if the last object is exception,  but I'm not sure if it works for your case.

File: alpha/alpha-fsm/src/main/java/org/apache/servicecomb/pack/alpha/fsm/channel/redis/MessagePublisher.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package org.apache.servicecomb.pack.alpha.fsm.channel.redis;
```
Comment (WillemJiang) [inline alpha/alpha-fsm/src/main/java/org/apache/servicecomb/pack/alpha/fsm/channel/redis/MessagePublisher.java]: License header.

File: pom.xml [hunk 3]
```diff
@@ -83,6 +83,7 @@
     <netty.boringssl.version>2.0.7.Final</netty.boringssl.version>
     <netty.version>4.1.24.Final</netty.version>
     <zookeeper.version>3.4.13</zookeeper.version>
+    <fastjson.version>1.2.58</fastjson.version>
```
Comment (WillemJiang) [inline pom.xml]: If we add new dependency, we need to update the license file [here](https://github.com/apache/servicecomb-pack/blob/master/distribution/src/release/LICENSE)

File: alpha/alpha-fsm/src/main/java/org/apache/servicecomb/pack/alpha/fsm/channel/RedisActorEventChannel.java [hunk 4]
```diff
@@ -31,13 +33,17 @@
...
-      ActorEventSink actorEventSink, MetricsService metricsService) {
+      ActorEventSink actorEventSink, MetricsService metricsService, RedisMessagePublisher redisMessagePublisher) {
     super(actorEventSink, metricsService);
+    this.redisMessagePublisher = redisMessagePublisher;
   }
 
   @Override
   public void sendTo(BaseEvent event){
-    throw new UnsupportedOperationException("Doesn't implement yet!");
+    LOG.info("sendTo message = [{}]",event);
```
Comment (coolbeevip) [inline alpha/alpha-fsm/src/main/java/org/apache/servicecomb/pack/alpha/fsm/channel/RedisActorEventChannel.java]: I suggest using LOG.debug

File: alpha/alpha-fsm/src/main/java/org/apache/servicecomb/pack/alpha/fsm/channel/redis/RedisMessagePublisher.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    private ChannelTopic channelTopic;
+
+    public RedisMessagePublisher(RedisTemplate<String, Object> redisTemplate, ChannelTopic channelTopic) {
+        this.redisTemplate = redisTemplate;
+        this.channelTopic = channelTopic;
+    }
+
+    @Override
+    public void publish(Object data) {
+        logger.info("send message [{}] to [{}]", data, channelTopic.getTopic());
```
Comment (coolbeevip) [inline alpha/alpha-fsm/src/main/java/org/apache/servicecomb/pack/alpha/fsm/channel/redis/RedisMessagePublisher.java]: I suggest using 
```java
if(LOG.isDebugEnabled()){
  LOG.debug(xxx)
}
```

File: alpha/alpha-fsm/src/main/java/org/apache/servicecomb/pack/alpha/fsm/channel/AbstractActorEventChannel.java (hunk 1 above)
Comment (coolbeevip) [reply alpha/alpha-fsm/src/main/java/org/apache/servicecomb/pack/alpha/fsm/channel/AbstractActorEventChannel.java]: Which reminds me, here should throw an exception

**Comment by coolbeevip** [comment]
When Alpha cluster, only one RedisMessageSubscriber should be allowed at the same time. Otherwise, it will repeat consumption. 

You can use the following method to determine whether the current node is the master node.

```java
@Autowride
NodeStatus nodeStatus;

if(nodeStatus.isMaster()){
  // Is master
}
```

File: alpha/alpha-fsm/src/main/java/org/apache/servicecomb/pack/alpha/fsm/channel/redis/RedisConfiguration.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
+package org.apache.servicecomb.pack.alpha.fsm.channel.redis;
```
Comment (WillemJiang) [inline alpha/alpha-fsm/src/main/java/org/apache/servicecomb/pack/alpha/fsm/channel/redis/RedisConfiguration.java]: We need to add License header here.

**Pull request closed without merging**